indexer/packages/postgres/src/stores/pnl-ticks-table.ts

```diff
@@ -268,3 +290,159 @@ export async function findMostRecentPnlTickForEachAccount(
     'subaccountId',
   );
 }
+
+export async function getRankedPnlTicks(
+  timeSpan: string,
+): Promise<LeaderboardPnlCreateObject[]> {
+  if (timeSpan === 'ALL_TIME') {
+    return getLatestRankedPnlTicks();
+  }
+  return getRankedPnlTicksForTimeSpan(timeSpan);
+}
+
+async function getRankedPnlTicksForTimeSpan(
+  timeSpan: string,
+): Promise<LeaderboardPnlCreateObject[]> {
+  const vaultAddresses: string[] = getVaultAddresses();
+  const vaultAddressesString: string = vaultAddresses.map((address) => `'${address}'`).join(',');
+  const intervalSqlString: string = convertTimespanToSQL(timeSpan);
+  const result: {
+    rows: LeaderboardPnlCreateObject[]
+  } = await knexReadReplica.getConnection().raw(
+    `
+    WITH latest_subaccount_pnl_x_days_ago_ranked AS (
+      SELECT
+          a."subaccountId",
+          a."totalPnl",
+          b."address",
+          ROW_NUMBER() OVER (PARTITION BY a."subaccountId" ORDER BY a."blockHeight" DESC) AS "rn"
+      FROM
+          pnl_ticks a
+      LEFT JOIN
+          subaccounts b ON a."subaccountId" = b."id"
+      WHERE
+          a."createdAt"::date <= (CURRENT_DATE - INTERVAL '${intervalSqlString}')
```

Comment:
If we're selecting a subaccount that was created 3 days ago and we're getting the 7 days ago ranked, it seems like we wouldn't get a result for them. I think we should sync with Yi on if that's the product outcome we want

Reply:
We do get a results for them if you look below `a."totalPnl" - COALESCE(b."totalPnl", 0) as "pnlDifference",` would just give latest pnl(which is for the last 3 days)